### PR TITLE
Remove ActiveSupport

### PR DESCRIPTION
**Pull request by jhass**
Depend on hashery for a AS::HWIA replacement. Replace all other
usages with core behavior.

ActiveSupport is an unnecessary big dependency. `json-jwt` is very useful outside Rails contexts, too. Bloating the dependency tree of every project directly or indirectly depending on it seems unreasonable for the "benefits" we get.

### CODE REVIEW

File: lib/json/jws.rb [hunk 1]
```diff
@@ -112,9 +114,10 @@ def with_jwk_support(key)
       when JSON::JWK
         key.to_key
       when JSON::JWK::Set
-        key.detect do |jwk|
-          jwk[:kid] && jwk[:kid] == kid
-        end.try(:to_key) or raise JWK::Set::KidNotFound
+        jwk = key.detect(proc { raise JWK::Set::KidNotFound }) do |jwk|
+          jwk[:kid] && jwk[:kid] == kid && jwk.respond_to?(:to_key)
+        end
```
Comment (baweaver) [inline lib/json/jws.rb]: I'd almost argue the `or raise` was clearer than `find` / `detect` taking a second block for default execution.

**Comment by jhass** [comment]
@nov Ping, any word on whether this has any prospect of getting in?

**Comment by jhass** [comment]
Updated.

**Comment by acorncom** [comment]
Any word here, I'd love to see this get merged as well ...

**Comment by kyledrake** [comment]
+1 to this. It's a show stopper for me for letsencrypt integration. It's doing "fun things" to my code base.

cancer

**Pull request closed without merging**